feat(Makefile): Default sanity checks for all repos

owlot opened the pull request:
Refactored the Makefile to split up in separate includes so they can be reused for different repository types.

This should also solve the issue that sometimes files with the wrong formatting was added through Terraform provisioning, which broke the pipelines in other projects.

ServiceDog commented on src/repositories_rust.tf:
```diff
@@ -44,12 +44,18 @@ locals {
+      ".make/toml.mk" = {
+        content = file("templates/all/.make/toml.mk")
       },
       ".taplo.toml" = {
```
cspell and editorconfig .mk?

owlot replied:
they are part of the default file list: https://github.com/Arrow-air/tf-github/pull/40/files#diff-1956abf05e8db0150fc12428093f137d1e9cca0a1d2d780a890c054274fb1e30R22 which will be merged into this

ServiceDog commented on Makefile:
```diff
@@ -0,0 +1,14 @@
+
+DOCKER_NAME := arrow-${name}
+
+help: .help-base .help-cspell .help-markdown .help-editorconfig
```
toml, python, docker helps and includes?

owlot replied:
I don't think they'll be needed for the tf-github repo, although, maybe toml would be helpful to check here too

ServiceDog commented on src/repositories_templates.tf:
```diff
@@ -108,7 +108,7 @@ module "repository_svc_template" {
     try(each.value.files, {}),
     { for file, path in local.template_files :
-      file => { content = templatefile(path, { owner_team = each.value.owner_team }) }
+      file => { content = templatefile(path, { owner_team = each.value.owner_team, name = format("lib-template-%s", each.key) }) }
```
*svc-template

owlot replied:
fixed bf62efa

ServiceDog commented on src/templates/all/Makefile:
```diff
@@ -0,0 +1,14 @@
+
+DOCKER_NAME := arrow-${name}
+
+help: .help-base .help-cspell .help-markdown .help-editorconfig
```
toml, python, rust, docker helps and includes?

owlot replied:
This is the 'default' Makefile, it's the minimal that will be deployed to all repo's.
Overrides will be there if more tests are needed (see the Makefile.tftpl specific for rust).

ServiceDog commented on src/templates/rust-all/Makefile.tftpl:
```diff
@@ -1,207 +1,36 @@
+include .make/docker.mk
 %{ endif }
+
+test: rust-check rust-test rust-clippy rust-fmt toml-test python-test md-test-links
```
cspell and editorconfig?

owlot replied:
fixed bf62efa

ServiceDog commented on src/templates/rust-all/.github/workflows/generate_docs.yml:
```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, and run hardhat dodoc on Arrow contracts
```
This might be better in its own PR

owlot replied:
removed from the pr

owlot commented:
@ServiceDog I added 2 more commits. I noticed I hadn't merged in the local.files map for all repo's created in main.tf and I added a fix for the branch protections which should resolve the issue where you were able to push to personal branches without signing commits.
